lib/pdk/util.rb

```diff
@@ -1,8 +1,38 @@
 require 'tmpdir'
 require 'tempfile'
+require 'fiddle/import' if Gem.win_platform?
 
 module PDK
   module Util
+    module WindowsFunctions
+      if Gem.win_platform?
+        extend Fiddle::Importer
+        dlload 'kernel32.dll'
+        extern('int GetLongPathNameA(char*, char*, int)', :stdcall)
+
+        # Call into kernel32.dll to resolve short path e.g. C:\Users\~ADMIN1\foo
+        # should be changed to long path i.e. C:\Users\Administrator\foo
+        def long_path(short_path)
+          # Fetch long path name into a buffer
+          bufferlen = 255
+          buffer    = nil
+          loop do
+            buffer = ' ' * bufferlen
+            len    = GetLongPathNameA(short_path.to_s, buffer, buffer.size)
+            # In the case of the path name being exceptionally long; resize the buffer
+            # to the actual size, then re-get it
+            if bufferlen < len
```

Comment:
The correct way to get the buffer length is to call `GetLongPathName` with a buffer size of zero, which will then return the required buffer size.  Then you can allocate the buffer and call with the correct size.